design/foundation-proposal.md

```diff
@@ -0,0 +1,135 @@
+  interested in collaborating on bare metal host provisioning with Kubernetes.
+
+### Non-Goals
+
+* No changes to the metal3-io code are in scope of this proposal.  This
+  proposal is only about the community future.
+* No changes to committer processes are proposed here.  We already have a
+  documented [process for adding and removing
+  maintainers](../maintainers/README.md).
+
+## Proposal
+
+Apply to become a CNCF Sandbox project.
+
+https://www.cncf.io/sandbox-projects/
+
+The [Goals](#goals) section discusses some of the immediate changes we should
+expect, but as the project continues to mature, many of the other services
+provided by the CNCF would be very beneficial to metal3-io.  The CNCF web site
+does a nice job discussing what services they offer projects on the
+[Services for CNCF Projects](https://www.cncf.io/services-for-projects/) page.
+
+## Design Details
+
+### Work Items
+
+* (russellb) Write this proposal.
+* (everyone) Provide feedback on initial proposal, reach consensus.
+* (russellb) If approved, draft the [CNCF Sandbox
+  application](https://github.com/cncf/toc/blob/master/process/project_proposals.adoc)
+  first as a draft in this repository.
+* (everyone) Provide feedback on application text, reach consensus.
+* (russellb) Submit Application.
+* (russellb + others) Prepare and deliver a presentation to the CNCF TOC.
+  Invite metal3 community members to attend.
+* (russellb) If accepted, transfer project assets as appropriate to CNCF.
+
+## Alternatives
+
+### No Foundation
+
+We could stick with the status quo, where one person or organization owns
+certain assets (like domains), and evolve our governance as needed to support
+the collaboration among many different teams at different organizations.  While
+we seem to be working well together so far, we would also miss out on some of
+the benefits of being more closely aligned with the CNCF ecosystem.
+
+### OSF (OpenStack Foundation)
+
+The baremetal-operator makes use of Ironic, which came out of OpenStack.
+Ironic is hidden as an implementation detail, though.  In terms of where we
+expect the metal3 components to be used, it’s really more closely aligned with
+the Kubernetes ecosystem and the CNCF.
+
+### Kubernetes Cluster Lifecycle SIG
+
+One of the components of metal3-io is a provider for
+[cluster-api](https://github.com/kubernetes-sigs/cluster-api).  While the
+provider itself could fall under this SIG, the scope of metal3-io is larger
+than supporting cluster-api.  metal3-io provides a Kubernetes API for generic
+bare metal host provisioning and cluster-api integration is just one use case.
+Since the scope of metal3-io is more broadly about bare metal host
+provisioning, we decided to organize it as an independent project.  It’s also
+convenient to keep the cluster-api provider along side the other related git
+repositories as part of metal3-io.  metal3-io community members do continue to
+collaborate with the cluster-api project.
```

Comment:
well stated Cluster API SIG is the first use-case and important one. But not the only use case.
Do we need to come up with a tentative list of use cases? Or that is not needed for CNCF sandbox application?